Source/EasyNetQ/Consumer/HandlerCollection.cs

```diff
@@ -71,13 +67,15 @@ public HandlerCollection(IEasyNetQLogger logger)
             return (message, info) => Task.Factory.StartNew(() => { });
         }
 
-        public dynamic GetHandler(Type messageType)
+        public Func<IMessage<object>, MessageReceivedInfo, Task> GetHandler(Type messageType)
         {
             Preconditions.CheckNotNull(messageType, "messageType");
-
+            
             var getHandlerGenericMethod = GetType().GetMethod("GetHandler", new Type[0]).MakeGenericMethod(messageType);
+            
+            var func = (Delegate)getHandlerGenericMethod.Invoke(this, new object[0]);
```

Comment:
You probably want to go `Invoke(this, null)` here, I think it's best to not instantiate an empty array when we don't need it since the target method doesn't take any parameter.